app/uk/gov/hmrc/soletraderidentificationfrontend/utils/MessagesHelper.scala

```diff
@@ -18,12 +18,44 @@ package uk.gov.hmrc.soletraderidentificationfrontend.utils
 
 import play.api.http.HttpConfiguration
 import play.api.i18n.{DefaultMessagesApi, DefaultMessagesApiProvider, Langs}
-import play.api.libs.json.Json
 import play.api.{Configuration, Environment}
-import uk.gov.hmrc.soletraderidentificationfrontend.models.{JourneyConfig, TranslationLabels}
+import uk.gov.hmrc.soletraderidentificationfrontend.models.{JourneyConfig, JourneyLabels}
 
 import javax.inject.{Inject, Singleton}
 
+object MessagesHelper {
+
+  private val optServiceNameTranslationKey = "optServiceName"
+  private val optFullNameTranslationKey = "optFullNamePageLabel"
+
+  def amendMessagesWithLabelsFromJourneyConfig(initialMessages: Map[String, Map[String, String]],
+                                               journeyConfig: JourneyConfig): Map[String, Map[String, String]] = {
+
+    val extraEnglishTranslations: Map[String, String] = journeyConfig
+      .pageConfig
+      .labels match {
+      case Some(JourneyLabels(_, Some(optEnglishServiceName), _, Some(optEnlgishFullName))) => Map(optServiceNameTranslationKey -> optEnglishServiceName, optFullNameTranslationKey -> optEnlgishFullName)
```

Comment:
this case do not match the order of JourneyLabels atm. which is JourneyLabels(welsh,english,english,welsh) and also what if service name is provided and fullName label not? there is no case for this situation (for welsh below as well)